src/Microsoft.SqlTools.ServiceLayer/FileBrowser/FileBrowserOperation.cs

```diff
@@ -160,7 +160,19 @@ public void ExpandSelectedNode(string expandPath)
                         {
                             this.fileTree.SelectedNode = lastNode;
                         }
-                        throw new FileBrowserException(string.Format(SR.InvalidPathError, this.expandPath));
+
+                        // If root folders returned from sys.dm_os_enumerate_filesystem don't match the folders present, expandPath might not be found
+                        // Bug: https://github.com/microsoft/azuredatastudio/issues/4767
+                        // Workaround : scope file browser tree to expandPath explicitly 
+                        try
+                        {
+                            this.ScopeFileTreeToPath(expandPath);
+                            return;
+                        }
+                        catch
```

Comment:
What is the type of Exception that we are replacing with the FileBrowserException? Should we wrap the exception caught as an inner exception of log the details of the exception that we are swallowing?

Reply:
Yes that would be good idea, however in file browser service the result only seems to pick the message from exception as result.Message = ex.Message; so it wouldn't matter. I can take the change of enabling passing on the whole exception as a separate change overall.